tests/datasets/test_dataset_factory.py

```diff
@@ -81,7 +49,18 @@ def test_disabled(disable_datasets: Iterator[None]) -> None:
         get_dataset("events")
 
 
+def test_get_dataset_multithreaded_collision() -> None:
```

Comment:
I don't think this is testing the same thing as the original test. The original test was ensuring that multiple threads initializing the factory at the same time wouldn't cause any issues.

Reply:
Right, this does not explicitly test with multiple threads, but it ensures that the factory object initialization is not decoupled from the entity mapping creation. If atomicity is guaranteed between these operations, then multiple thread should have no issues accessing the factory in concurrently.

Reply:
OK. Can you change the test name then please?